LibGit2/Functions/Send-GitCommit.ps1

```diff
@@ -0,0 +1,89 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function Send-GitCommit

```

Comment:
This function should return an enumeration value based on what happened:

````
public enum PushResult
{
    Ok,
    Failed,
    Rejected
}
```

The way this is currently written, any automation that tries to push will fail when there are incoming changes. Callers need to know if the push is something they can handle. In the case where the push is rejected because there are new remote changes, automation can use that information to pull those changes down, then re-attempt the push.

Reply:
Added output enumerator.